### PR TITLE
Some errorprone mitigations (no functional change)

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/options/EnvironmentDefaultProvider.java [hunk 1]
```diff
@@ -38,7 +39,7 @@ public String defaultValue(ArgSpec argSpec) {
       OptionSpec optionSpec = (OptionSpec) argSpec;
       return Arrays.stream(optionSpec.names())
           .filter(name -> name.startsWith("--"))
-          .map(name -> "NESSIE_GC_" + name.substring(2).replace('-', '_').toUpperCase())
+          .map(name -> "NESSIE_GC_" + name.substring(2).replace('-', '_').toUpperCase(Locale.ROOT))
```
Comment (XN137) [inline gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/options/EnvironmentDefaultProvider.java:42]: should we set the new upper/lowercase locale check to ERROR ?

File: gc/gc-repository-jdbc/src/test/java/org/projectnessie/gc/contents/jdbc/TestJdbcPersistenceUnit.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 public class TestJdbcPersistenceUnit {
   @Test
+  @SuppressWarnings("InlineMeInliner")
```
Comment (XN137) [inline gc/gc-repository-jdbc/src/test/java/org/projectnessie/gc/contents/jdbc/TestJdbcPersistenceUnit.java:27]: if this code is considered java11 then we should use the new api here instead probably

**Review by XN137 — APPROVED** [review]

File: gc/gc-repository-jdbc/src/test/java/org/projectnessie/gc/contents/jdbc/TestJdbcPersistenceUnit.java (hunk 2 above)
Comment (snazy) [reply gc/gc-repository-jdbc/src/test/java/org/projectnessie/gc/contents/jdbc/TestJdbcPersistenceUnit.java:27]: Yea, but it's just a test

File: gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/options/EnvironmentDefaultProvider.java (hunk 1 above)
Comment (snazy) [reply gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/options/EnvironmentDefaultProvider.java:42]: Mind opening a PR?

**Pull request merged**